### PR TITLE
clarify how  `predictions_to_labeled_instances` work for targeted or non-targeted hotflip attack

### PR DESCRIPTION
**Pull request by XinzheL**
Fixes # .

Changes proposed in this pull request:

- fix docs for `Predictor.predictions_to_labeled_instances` so to cover its use in the attack scenario 
-

### CODE REVIEW

File: allennlp/predictors/predictor.py [hunk 1]
```diff
@@ -268,12 +268,13 @@ def predictions_to_labeled_instances(
...
-        to the output. For example, in classification this function labels the instance according
-        to the class with the highest probability. This function is used to to compute gradients
-        of what the model predicted. The return type is a list because in some tasks there are
-        multiple predictions in the output (e.g., in NER a model predicts multiple spans). In this
-        case, each instance in the returned list of Instances contains an individual
-        entity prediction as the label.
+        to the `outputs`. This function is used to (1) compute gradients of what the model predicted; 
+        (2) label the instance for the attack. For example, (a) for the untargeted attack for classification 
+        this function labels the instance according to the class with the highest probability; (b) for 
+        targeted attack, it directly constructs fields from given target.
```
Comment (epwalsh) [inline allennlp/predictors/predictor.py:274]:
```suggestion
        targeted attack, it directly constructs fields from the given target. 
```

**Review by epwalsh — APPROVED** [review]
Thanks @XinzheL!

**Comment by epwalsh** [comment]
Looks like there are some minor formatting issues (trailing whitespace). Can you just run `black .`? That should fix it.

**Comment by XinzheL** [comment]
> Looks like there are some minor formatting issues (trailing whitespace). Can you just run `black .`? That should fix it.

Thank you for the instruction. I've done that.

**Comment by leo-liuzy** [comment]
Also, just a reminder, please document your changes to CHANGELOG.md (1-2 sentences are enough).

**Pull request merged**